sympy/matrices/dense.py

```diff
@@ -171,39 +171,27 @@ def _eval_extract(self, rowsList, colsList):
         return self._new(len(rowsList), len(colsList),
                          list(mat[i] for i in indices), copy=False)
 
-    def _eval_matrix_mul(self, other):
-        from sympy import Add
+    def _eval_matrix_mul(self, other, mulsimp=None):
         # cache attributes for faster access
-        self_rows, self_cols = self.rows, self.cols
+        self_rows, self_cols   = self.rows, self.cols
         other_rows, other_cols = other.rows, other.cols
-        other_len = other_rows * other_cols
-        new_mat_rows = self.rows
-        new_mat_cols = other.cols
-
-        # preallocate the array
-        new_mat = [self.zero]*new_mat_rows*new_mat_cols
+        other_len              = other_rows * other_cols
+        new_mat_rows           = self_rows
+        new_mat_cols           = other_cols
+        self_mat               = self._mat
+        other_mat              = other._mat
+        new_mat                = [self.zero]*new_mat_rows*new_mat_cols
```

Comment:
You have a strange style of indentation. Please use the PEP8ish style that is already used in SymPy i.e. in the above 1 space before and after `=`.

Reply:
Actually the assignment above seem unnecessary. Have you measured that any of them has any appreciable impact on performance?